doc/examples/rendering/wedgeTests.gfr

```diff
@@ -413,54 +412,47 @@ __children["Outputs"]["outputs"]["output1"]["fileName"].setValue( '${project:roo
+__children["Backdrop10"]["__uiBound"].setValue( imath.Box2f( imath.V2f( -13.9212952, -17.67 ), imath.V2f( 43.3652649, 10.0430069 ) ) )
 __children["Backdrop10"]["__uiPosition"].setValue( imath.V2f( -18.73316, -64.9413757 ) )
-Gaffer.Metadata.registerValue( __children["Backdrop11"], 'nodeGadget:color', imath.Color3f( 0.368000001, 0.400000006, 0.359999985 ) )
-__children["Backdrop11"]["title"].setValue( 'Parameter Value Burn-in' )
-__children["Backdrop11"]["description"].setValue( "It's often convenient when reviewing Wedge renders to know what value was used for each frame. We can make use of Gaffer's image nodes to burn in some text of the parameter name and value onto the renders.\n\nThis demonstrates how, even when the type of data flowing through the graph changes, the context in which it is evaluated in always ripples up from the node being executed. This allows the context variables set by the downstream Wedge node to control multiple aspects of the render -> composite process at once. The Text node references the same variable as the shader parameter plug's expression, so it will always be in sync.\n\nThe execution nodes in Gaffer, which are typically red, are task nodes. These can be chained together to form a series of operations. So, when the Wedge node executes, Gaffer will have made sure the ImageWriter and preceding AppleseedRender node have already been executed, in the requested context." )
-__children["Backdrop11"]["__uiBound"].setValue( imath.Box2f( imath.V2f( 4.16482925, -17.3254013 ), imath.V2f( 63.6761055, 11.3694763 ) ) )
+__children["Backdrop11"]["title"].setValue( 'Parameter value burn-in' )
+__children["Backdrop11"]["description"].setValue( "It's common practice to annotate each wedged render with the value of the ramped value, to make review easier. Here we use some of Gaffer's image nodes to burn in annotations that list the shader parameter name and value onto each render.\n\nThe red nodes in Gaffer are task nodes. When you execute a graph, you're actually executing one or more task nodes. When chained together, they execute in series, starting with the earliest in the chain. This way, you can arrange for a string of operations (such as a render, write, or system command) to occur before the last task node is executed. In this graph, when you execute theWedge node, the upstream AppleseedRender, ImageWriter, and SystemCommand node will execute beforehand, in that order.\n\nLike before, we make use of the `wedge:metallic` context variables pushed by the downstream Wedge node. This time, the Text node inserts the value into it in its string. Since this will be the same value as the shader parameter plug that we drive upstream, the two will always be in sync." )
```

Comment:
0b9a08c6848b9864fe4d6d69e90423105c5000db

Reply:
The call-out is great, thanks.

> ...  somewhat **inintuitive**, but... 

unintuitive?

> ... the graph, passing up from node to node, child to parent.

Could we add something at the end along the lines of "even when the type of data between the nodes changes".  This allows us to make use of the wedge node context variables in both 2D image processing and 3D Scene rendering - as they are connected via task dependencies."?

I think the key thing to help people understand in specific terms, is that the otherwise separate 2D graph that does the compositing and the 3D rendering graph share the context as of the task connections. This is quite unique to Gaffer.

Reply:
109b7173d782b15797eed92a6237f2c7751f93c7